### PR TITLE
Use Complement libraries

### PR DESCRIPTION
**Pull request by kegsay**
An experiement in using bits of Complement. Doesn't use all the machinery, as we don't deploy new containers for each test, but it's a start (uses must/match/client).

### CODE REVIEW

**Comment by DMRobertson** [comment]
> Doesn't use all the machinery, as we don't deploy new containers for each test, but it's a start (uses must/match/client).

Do we even want to do this? While we do have tests that leak state, the speed of running them locally is really valuable and it would be sad to have to wait for Synapse's docker container to start up over and over again.

File: tests-e2e/client_test.go [hunk 1]
```diff
@@ -38,635 +26,47 @@ type Event struct {
...
-// This function should not be used to set an "access_token" parameter for Matrix authentication.
-// Instead, set CSAPI.AccessToken.
-func WithQueries(q url.Values) RequestOpt {
-	return func(req *http.Request) {
-		req.URL.RawQuery = q.Encode()
-	}
+	*client.CSAPI
+	Localpart string
+	Domain    string
+	AvatarURL string
```
Comment (DMRobertson) [inline tests-e2e/client_test.go:43]: Should this be folded into complement?

**Review by DMRobertson — APPROVED** [review]

File: tests-e2e/client_test.go (hunk 1 above)
Comment (kegsay) [reply tests-e2e/client_test.go:43]: Maybe. Unsure for now, so leaving it here.

**Comment by kegsay** [comment]
> Do we even want to do this? 

Maybe in the future when we reuse containers. Right now, no.

**Pull request merged**